recurly/base_client.py

```diff
@@ -25,9 +25,10 @@ def request_converter(value):
 
 
 class BaseClient:
-    def __init__(self, api_key):
+    def __init__(self, api_key, timeout=None):
         self.__api_key = api_key
-        self.__conn = http.client.HTTPSConnection(HOST, PORT)
+        self.timeout = timeout if timeout is not None else DEFAULT_REQUEST_TIMEOUT
```

Comment:
I worry it might be a little misleading to make this a settable property on the client. A programmer may think they can change this. Example:

```python
client = Client('key', timeout=10)
client.do_something()
client.timeout = 100
client.do_something_that_takes_longer()
```

If we want to expose this, would there be a way to make a `timeout()` method on the `Client` perhaps that just reads the property from `__conn`?

Reply:
Good point, there's really no reason to expose it via `self.timeout`. I think it's best to remove `self.timeout`. If someone wants to change it, they can construct a new client.

Reply:
This has been updated in the latest commit. I remembered that I used `self.timeout` to make testing easier, but I can still access `__conn` in the test to verify timeout is being set properly.